modules/electrophysiology_uploader/php/upload.class.inc

```diff
@@ -0,0 +1,249 @@
+        $upload_dir    = new \SplFileInfo(
+            $config->getSetting('EEGUploadIncomingPath')
+        );
+
+        $downloader = new \LORIS\FilesDownloadHandler(
+            new \SPLFileInfo($upload_dir . '/' . $file_rel_path)
+        );
+
+        return $downloader->handle(
+            $request->withAttribute('filename', $filename)
+        );
+    }
+
+    /**
+     * Processes the values & saves to database and return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request) : ResponseInterface
+    {
+        $db = $this->loris->getDatabaseConnection();
+
+        $uploadedFile = $request->getUploadedFiles()['eegFile'] ?? null;
+        $values       = $request->getParsedBody();
+        $pscid        = $values['pscid'] ?? null;
+        $candID       = $values['candID'] ?? null;
+        $visitLabel   = $values['visit'] ?? null;
+
+        if (empty($request->getUploadedFiles()) && empty($values)) {
+            $upload_max_size = \Utility::getMaxUploadSize();
+            return new \LORIS\Http\Response\JSON\InternalServerError(
+                "The uploaded file size ($upload_max_size) exceeds " .
+                "the upload_max_filesize or post_max_size directive in php.ini"
+            );
+        }
+
+        if (is_null($uploadedFile)) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'No file uploaded'
+            );
+        }
+
+        $sessionID = $db->pselectOne(
+            "
+            SELECT s.ID
+            FROM session s
+                LEFT JOIN candidate c ON (c.CandID = s.CandID)
+            WHERE s.Visit_label = :visit
+                AND s.CandID = :candID
+                AND c.PSCID = :pscid
+                ",
+            [
+                'visit'  => $visitLabel,
+                'candID' => $candID,
+                'pscid'  => $pscid
+            ]
+        );
+
+        if (is_null($sessionID)) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Visit for candidate does not exist'
+            );
+        }
+
+        $config   = \NDB_Config::singleton();
+        $filename = urldecode($uploadedFile->getClientFilename());
+
+        if ($filename !== "{$pscid}_{$candID}_{$visitLabel}_bids.tar.gz") {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Filename does not match expected name, please verify.'
+            );
+        }
+
+        $targetdir = new \SplFileInfo($config->getSetting('EEGUploadIncomingPath'));
+        try {
+            $uploader = (new \LORIS\FilesUploadHandler($targetdir));
+        } catch (\Exception $e) {
+            return new \LORIS\Http\Response\JSON\InternalServerError(
+                $e->getMessage()
+            );
+        }
+
+        // Check for existing files
```

Comment:
Why is it checking for existing files? What is it checking for? What is it doing if it finds them? I'm confused about this part of the code because the FilesUploadHandler already checks if files exist (and either returns an error or overwrites depending on the withOverwrite setting in it.)